src/librustc_middle/mir/interpret/error.rs

```diff
@@ -358,7 +358,10 @@ pub enum UndefinedBehaviorInfo {
     /// Using a non-character `u32` as character.
     InvalidChar(u32),
     /// Using uninitialized data where it is not allowed.
-    InvalidUndefBytes(Option<Pointer>),
+    InvalidUndefBytes {
+        ptr: Option<Pointer>,
```

Comment:
Since these are always either both `None` or both `Some`, please create a separate datastructure holding the `Pointer` and the `Size` and use only a single `Option`.

Reply:
Or maybe just use a tuple? Not sure if it is worth naming that.